daemon/cluster/cluster.go

```diff
@@ -725,6 +730,21 @@ func (c *Cluster) UpdateService(serviceID string, version uint64, spec types.Ser
 		return err
 	}
 
+	if encodedAuth != "" {
+		serviceSpec.Task.Runtime.(*swarmapi.TaskSpec_Container).Container.PullOptions = &swarmapi.ContainerSpec_PullOptions{RegistryAuth: encodedAuth}
+	} else {
+		// this is needed because if the encodedAuth isn't being updated then we
+		// shouldn't lose it, and continue to use the one that was already present
+		currentService, err := getService(c.getRequestContext(), c.client, serviceID)
+		if err != nil {
+			return err
+		}
+		ctnr := currentService.Spec.Task.Runtime.(*swarmapi.TaskSpec_Container).Container
```

Comment:
Let's do this as follows:

```go
ctnr := currentService.Spec.Task.GetContainer()
if cntr == nil {
        return errors.New("service does not use container tasks")
}
```

Going forward, there's no guarantee that a service will use containers. We don't want to panic if we encounter a service like this.